### ISSUE TITLE
Realtime inference

### ISSUE THREAD

**Issue by Siziff** [JOY-MM/JoyGen#11]
@gobigrassland  Do I understand correctly that this project was created based on MuseTalk ideas? I saw a lot of your comments in this discussion https://github.com/TMElyralab/MuseTalk/issues/81 .  Some questions that interest me:
1. Is it also possible, as in MuseTalk, to inference the model in "Real-time" while doing data preprocessing?
2. Is it possible to use the training code from this repository on MuseTalk base model?
3. Is there are some methods to increase model inference time?? 

Thanks a lot for the project, it is very useful and interesting :)

**Comment by LearnedVector** [comment]
i would also like the know the above ^, great project btw

**Comment by gobigrassland** [comment]
@Siziff   We are building upon MuseTalk to generate talking-face videos with audio-to-mouth shape matching, guided by explicit mouth depth maps. There are still significant differences between our approach and MuseTalk. Our paper emphasizes how to achieve precise audio-to-lip synchronization.
1. The inference process includes additional steps for audio-to-motion conversion and 3D reconstruction, which makes our approach more time-consuming compared to MuseTalk.
2. No, the input channel number of the U-Net is 12, not 8.
3. Sorry，we don't have time to think about this issue at the moment.

**Comment by Siziff** [comment]
@gobigrassland Thanks a lot for the quick reply :) About the first question, is it possible to cache latent transformations and masks for one video to quickly overlay different audio tracks on the same video?  

It is also very interesting, have you thought or tried to train lip-sync models based on flow-matching approaches instead of diffusion? Would this approach be faster and better?

**Comment by gobigrassland** [comment]
> [@gobigrassland](https://github.com/gobigrassland) Thanks a lot for the quick reply :) About the first question, is it possible to cache latent transformations and masks for one video to quickly overlay different audio tracks on the same video?
> 
> It is also very interesting, have you thought or tried to train lip-sync models based on flow-matching approaches instead of diffusion? Would this approach be faster and better?

1. Yes, it's possible. For the same video, both the results of 3D reconstruction and facial latent can be reused for different audio inputs. 
2. We previously tried training a lip-sync model based on the Wav2Lip method, but it was unreliable. If you can train a truly better lip-sync model, it might be possible to supervise the training with an auxiliary loss function, in which case additional mouth depth maps may not be needed.

**Comment by Siziff** [comment]
Thanks a lot for the answers and thanks again for the project 🥇
